adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/AmqpAdapterSaslAuthenticatorFactory.java

```diff
@@ -302,14 +302,15 @@ private void verifyExternal(final Handler<AsyncResult<DeviceUser>> completer) {
                             getCertificateAuthProvider().authenticate(credentials, currentSpan.context(), user);
                             return user;
                         })
+                        .compose(user -> checkTenantIsEnabled(tenantTracker.result())
```

Comment:
why is this check only done for SASL EXTERNAL?

Reply:
For SASL PLAIN, this check is also being done and it is part of the method `getTenantObject(credentials.getTenantId())`. 

For SASL EXTERNAL, the tenant object is first retrieved to get the `TrustAnchor` and thats why the check `checkTenantIsEnabled(tenantTracker.result()` is not part of the `getTenantObject(...)`.

Reply:
ahhh, now I see. This is very confusing FMPOV. Can you please align the behavior of the two methods so that they follow the same pattern?

Reply:
Done